src/main/java/ch/njol/skript/bukkitutil/ItemUtils.java

```diff
@@ -379,4 +380,40 @@ public static boolean isGlass(Material material) {
 				return false;
 		}
 	}
+
+	/**
+	 * Applies a provided {@code Consumer} to the meta of the provided {@code ItemStack} and returns the updated {@code ItemStack} (with updated {@code ItemMeta}).
+	 *
+	 * @param itemStack the item whose meta is to be changed using the provided Consumer
+	 * @param metaChanger a consumer to update the meta of the provided ItemStack
+	 * @param <T>
+	 * @return the updated item
+	 */
+	public static <T extends ItemMeta> ItemStack changeItemMeta(@NotNull ItemStack itemStack, @NotNull Consumer<T> metaChanger) {
+		//noinspection unchecked
+		T itemMeta = (T) itemStack.getItemMeta();
+		metaChanger.accept(itemMeta);
+		itemStack.setItemMeta(itemMeta);
+		return itemStack;
+	}
+
+	/**
+	 * Updates the provided object ({@code Slot}, {@code ItemType}, {@code ItemStack}) by setting it to the provided {@code ItemStack}.
+	 *
+	 * @param object the object to update
+	 * @param itemStack the item to set the object to
+	 * @see #asItemStack(Object)
+	 */
+	public static void setItem(Object object, @NotNull ItemStack itemStack) {
+		ItemMeta itemMeta = itemStack.getItemMeta();
+		if (object instanceof Slot slot) {
+			slot.setItem(itemStack);
+		} else if (object instanceof ItemType itemType) {
+			itemType.setItemMeta(itemMeta);
+		} else if (object instanceof ItemStack itemStack1) {
+			itemStack1.setItemMeta(itemMeta);
```

Comment:
Wait
we never actually change the item itself for these ones

Reply:
Wdym? There's no need to. The only thing that needs the item to be completely overridden/changed is `Slot` everything else is just changing the meta.

Reply:
The method is set item though
If i'm given a Dirt ItemStack and want to set it to a Cobblestone ItemStack, would I not use setItem?

Reply:
Well it was never intended to change it to a whole new item. You suggested the method name `setItem`. But I can change it to fully change the item if that's what you want.

Reply:
Also, what if my ItemType has multiple itemdatas in it? Do they all get removed? What if i want to check the itemmeta of all the datas, but keep the various types? I think there's a lot of nuance this method is ignoring right now.
It's acting more like `setItemMeta` than `setItem` for most inputs.

Reply:
> Well it was never intended to change it to a whole new item. You suggested the method name `setItem`. But I can change it to fully change the item if that's what you want.

I think it should be consistent, whatever it does. The easiest solution is to just make it only change the meta and name it setItemMeta. Up to you though.

Reply:
Done

Reply:
Well it has to change the behavior too
It should accept an itemmeta rather than a stack, and it shouldn't change the stack in the slot

Reply:
If I take out `Slot`, then that just beats the original purpose of this PR. This PR is meant to act as the setter from the getter of `#asItemStack`
There's no real way of setting the ItemMeta of the slot, because the only way to change it is by `#setItem`
What if I change  the  method name to `updateSource` or something

Reply:
Why would you take out slot? Just set the meta of the item in the slot? get it, change it, set it.

Reply:
fine, whatever  gets this approved